autorest/authorization.go

```diff
@@ -258,3 +261,21 @@ func (egta EventGridKeyAuthorizer) WithAuthorization() PrepareDecorator {
 	}
 	return NewAPIKeyAuthorizerWithHeaders(headers).WithAuthorization()
 }
+
+// BasicAuthorizer implements authorization for Basic authentication.
+type BasicAuthorizer struct {
+	token string
+}
+
+// NewBasicAuthorizer creates a new BasicAuthorizer with the specified token.
+func NewBasicAuthorizer(token string) *BasicAuthorizer {
```

Comment:
to avoid confusion, can we name this BasicAccessAuthorizer or BasicHttpAuthorizer ? to be more obvious it refers to the http basic auth and not some generic BasicAuthorizer

Reply:
We could but it wouldn't follow the pattern we have for other authorizers, e.g. `BearerAuthorizer`.  I will update the comment to be more clear.

Reply:
yea fair enough , to be fair the auth schemes are called "Basic" , "Bearer" ... . so yea maybe have the comment more explicit that it creates an authorizer using the Basic Http Authorization Scheme